proto/storage/auth_provider.proto

```diff
@@ -24,4 +24,14 @@ message AuthProvider {
     // not ui_endpoint.
     repeated string extra_ui_endpoints = 9;
     bool active                 = 10;
+
+    // RequiredAttribute allows to specify a set of attributes which are required to be returned
+    // by the auth provider. If required attributes are not set, the user will *not* be associated
+    // with any role and return access forbidden.
+    message RequiredAttribute {
+        string attribute_key   = 1;
+        string attribute_value = 2;
+        bool read_only         = 3;
+    }
+    repeated RequiredAttribute required_attributes = 11;
```

Comment:
Sorry if I missed this in the arch discussion, but why are we creating two kinds of "attributes": the existing `config` above, and this "config" with an additional property?

Reply:
Essentially what Ivan already mentioned: We wanted to avoid name-clashes with other config settings + if we decide to add multiple required attributes, it would be cumbersome to do something like `requiredAttribute1=value1,requiredAttribute2=value2`.